crates/bevy_animation/src/lib.rs

```diff
@@ -184,7 +196,7 @@ pub fn animation_player(
     children: Query<&Children>,
 ) {
     for (entity, mut player) in &mut animation_players {
-        if let Some(animation_clip) = animations.get(&player.animation_clip) {
+        if let Some(animation_clip) = player.animation_clip().and_then(|x| animations.get(x)) {
```

Comment:
I'd actually prefer for us to use `let ... else` with continue here instead of `and_then` inside the if expression.